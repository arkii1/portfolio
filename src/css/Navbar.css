.navbar {
    position: sticky;
    top: 0;
    background-color: var(--c-grey);
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 100vw;
    z-index: 10;
    box-shadow: 0 3px 5px 4px rgba(0, 0, 0, 0.2);
}

.navbar > button {
    font-size: 1rem;
    background-color: transparent;
    outline: none;
    border: none;
    color: var(--c-white);
    margin: 1rem 0;
    transition: 0.1s;
}

.navbar > button:hover {
    color: var(--c-blue);
    cursor: pointer;
    transform: scale(1.1);
}

.navbar > button:active {
    color: var(--c-white);
}