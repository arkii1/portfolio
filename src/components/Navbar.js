import React from "react"
import "../css/Navbar.css"

function Navbar() {
  const onHomeClick = () => {
    document.querySelector(".header").scrollIntoView({ behavior: "smooth" })
  }

  const onAboutClick = () => {
    document.querySelector(".about").scrollIntoView({ behavior: "smooth" })
  }

  const onProjectClick = () => {
    document.querySelector(".projects").scrollIntoView({ behavior: "smooth" })
  }

  const onContactClick = () => {
    document.querySelector(".contact").scrollIntoView({ behavior: "smooth" })
  }

  return (
    <nav className="navbar">
      <button type="button" onClick={onHomeClick}>
        Home
      </button>
      <button type="button" onClick={onAboutClick}>
        About
      </button>
      <button type="button" onClick={onProjectClick}>
        Projects
      </button>
      <button type="button" onClick={onContactClick}>
        Contact
      </button>
    </nav>
  )
}

export default Navbar
