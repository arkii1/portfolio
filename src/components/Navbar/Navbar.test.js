import React from "react"
import ReactDOM from "react-dom/client"
import { cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import Navbar from "./Navbar"

afterEach(cleanup)

it("renders without crashing", () => {
  const container = document.createElement("div")
  const root = ReactDOM.createRoot(container)
  root.render(<Navbar />)
})

it("matches snapshot", () => {
  const tree = renderer.create(<Navbar />).toJSON()
  expect(tree).toMatchSnapshot()
})
