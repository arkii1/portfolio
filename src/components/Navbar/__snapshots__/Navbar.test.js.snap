// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Home
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    About
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Projects
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Contact
  </button>
</nav>
`;
